internal/server/matches.go

```diff
@@ -25,6 +25,11 @@ func (s *Server) matchesHandler() http.HandlerFunc {
 
 		// Get new match data from TBA
 		if err := s.updateMatches(eventKey); err != nil {
+			// 404 if eventKey isn't a real event
+			if ok := store.IsNoResultError(err); ok {
```

Comment:
I would be in favor of this function being removed and replaced with `_, ok := err.(store.NoResultError)`. Also can you change `store.NoResultError` to just be `type ErrNoResults error`.